### ISSUE TITLE
Params should be a struct, not a map

### ISSUE THREAD

**Issue by rusenask** [mendableai/firecrawl-go#6]
Using just params makes it hard to understand what's available
```
func (app *FirecrawlApp) ScrapeURL(url string, params map[string]any) (*FirecrawlDocument, error) {
```

How about just put the whole thing like you already have in the API?
```go
type ScrapeRequest struct {
	URL         string `json:"url"`
	PageOptions struct {
		Headers struct {
		} `json:"headers"`
		IncludeHTML                      bool     `json:"includeHtml"`
		IncludeRawHTML                   bool     `json:"includeRawHtml"`
		OnlyIncludeTags                  []string `json:"onlyIncludeTags"`
		OnlyMainContent                  bool     `json:"onlyMainContent"`
		RemoveTags                       []string `json:"removeTags"`
		ReplaceAllPathsWithAbsolutePaths bool     `json:"replaceAllPathsWithAbsolutePaths"`
		Screenshot                       bool     `json:"screenshot"`
		FullPageScreenshot               bool     `json:"fullPageScreenshot"`
		WaitFor                          int      `json:"waitFor"`
	} `json:"pageOptions"`
	ExtractorOptions struct {
		Mode             string `json:"mode"`
		ExtractionPrompt string `json:"extractionPrompt"`
		ExtractionSchema struct {
		} `json:"extractionSchema"`
	} `json:"extractorOptions"`
	Timeout int `json:"timeout"`
}
```

```go
func (app *FirecrawlApp) ScrapeURL(req *ScrapeRequest) (*FirecrawlDocument, error) {
	headers := app.prepareHeaders("")
	...
```

Then you get autocompletion, suggestions for copilot, etc.

**Comment by rafaelsideguide** [comment]
@rusenask good suggestion. Working on that for v1.

**Comment by KentHsu** [comment]
Didn't see the param interfaces in js-sdk when working on the go-sdk. It will be great if we can update `Search` and `CrawlURL` methods too!

**Comment by rafaelsideguide** [comment]
Closed by #3
